### PR TITLE
Add dropdown context menu to schema editor fields

### PR DESCRIPTION
**Pull request by burieberry**
- Updated Boxel UI components with `DropdownButton` component from old Boxel package
- Added the button on schema editor and added a context menu with disabled options (there are separate tickets for each)
- Looks like this when items are not disabled:

contextmenu

### CODE REVIEW

**Review by FadhlanR — APPROVED** [review]

File: packages/boxel-ui/addon/components/dropdown-button/index.gts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+
+interface Signature {
+  Element: HTMLButtonElement;
+  Args: {
+    icon: string;
+    label: string;
+    class?: string;
+    contentClass?: string;
+    noHoverStyle?: boolean;
```
Comment (jurgenwerk) [inline packages/boxel-ui/addon/components/dropdown-button/index.gts]: What does this one mean?

File: packages/boxel-ui/addon/components/dropdown-button/index.gts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      </button>
+    </:trigger>
+    <:content as |dd|>
+      {{yield
+        (hash Menu=(component BoxelMenu closeMenu=dd.close) close=dd.close)
+      }}
+    </:content>
+  </BoxelDropdown>
+  <style>
+    @layer {
```
Comment (jurgenwerk) [inline packages/boxel-ui/addon/components/dropdown-button/index.gts:60]: What is the advantage of using @layer over adding a custom class to the dropdown and making rules depending on that?

Comment (burieberry) [reply packages/boxel-ui/addon/components/dropdown-button/index.gts:60]: `@layer` makes sure that if the custom class css and preexisting component css are conflicting, the custom class overrides the component css

File: packages/boxel-ui/addon/components/dropdown-button/index.gts (hunk 1 above)
Comment (burieberry) [reply packages/boxel-ui/addon/components/dropdown-button/index.gts]: I see that we've used this arg in the cardstack monorepo where we used this component, but it looks like it doesn't actually do anything. I'll remove it

**Pull request merged**